### DISCUSSION TITLE
DayPickerInput: with 2 months, clicking on the second (right-hand) month jumps the calendar forward

### DISCUSSION THREAD

**Discussion by HNBowen** [gpbl/react-day-picker#1118] [Support]
When render-ing the DayPickerInput with 2 months, selecting a day on the right-hand month advances the calendar forward a month.

This can be reproduced on the "Works with input fields" -> "range of days" example at http://react-day-picker.js.org/

Per discussion with @gpbl on #38  this may be caused by the `fromMonth` prop in `dayPickerProps` being ignored.

I'm investigating this issue!

**Comment by HNBowen** [comment]
@gpbl After a little digging I'm noticing a few key differences between the example I mentioned and my particular use case.

For one, the example is actually two separate DayPickerInputs while mine is a single DayPickerInput. I think the example behaves exactly as expected, in that the second DayPickerInput is meant to have it's `month` prop (the display month) match whatever month has been set by the user in the first DayPickerInput.

This simple codesandbox example is closer to my use case: https://codesandbox.io/s/4j9z8w4719. If you click on the second month, the calendar moves forward a month automatically. 

In contrast, what I'm hoping to get out of my single DayPickerInput is for the displayed `month` prop to not change when a user selects a day. The `month` prop is passed to the `DayPicker` in `DayPickerInput` via it's internal state, which is updated on day click here: https://github.com/gpbl/react-day-picker/blob/39ad48ecadb18355e28ee5ef1741ca484949760d/src/DayPickerInput.js#L500 

My use case may be an edge case (I can totally see why the displayed `month` automatically moves forward) but I do need to disable this default behavior.

What's the best way forward here? I'm not sure if this is something I should implement on my own fork of the library, or if having control over advancing the month in this fashion is something the broader community would have interest in.

**Comment by brojd** [comment]
@HNBowen The workaround is to make use of `onDayChange` and `ref` to set month:
```
    <DayPickerInput
            dayPickerProps={dayPickerProps}
            hideOnDayClick={false}
            ref={inputRef}
            component={component}
            onDayChange={() => {
              inputRef.current.setState({ month: new Date() }); // whatever date you want
            }}
          />
```
